ansible/roles/telegraf_systemd/tasks/install_telegraf.yml

```diff
@@ -1,18 +1,31 @@
 ---
 - name: Get influxdata apt repo key
   ansible.builtin.get_url:
-    url: https://repos.influxdata.com/influxdb.key
-    dest: /etc/apt/trusted.gpg.d/influxdb.asc
+    url: https://repos.influxdata.com/influxdata-archive_compat.key
+    dest: /etc/apt/keyrings/influxdata-archive_compat.pgp
     mode: 0644
     owner: root
     group: root
   environment: "{{ proxy_env | default('') }}"
+  register: get_url_output
+
+- name: dearmor pgp key to gpg key
+  ansible.builtin.shell:
+    cmd: cat /etc/apt/keyrings/influxdata-archive_compat.pgp | gpg --dearmor | tee {{ influxdata_apt_repo_key }}
+  when: get_url_output is changed
 
 - name: Install influxdata apt repo
+  ansible.builtin.apt_repository:
+    repo: "deb [arch=amd64 signed-by={{ influxdata_apt_repo_key }}] \
+      https://repos.influxdata.com/{{ ansible_lsb.id | lower }} {{ ansible_lsb.codename }} stable"
+    filename: influxdata-archive_compat
+    state: present
+
+- name: Remove old influxdata apt repo
```

Comment:
Maybe we've already discussed this in the past, but having a "delete old thing" kind of task feels a bit like clutter. Since we don't have users in the wild, I think we can just perform cleanup as a custom, one-time task.

Reply:
yep. Makes sense. I'll fix the linter errors and then also remove this.